libraries/botbuilder-ai/src/qnaMakerRecognizer.ts

```diff
@@ -261,15 +270,21 @@ export class QnAMakerRecognizer extends Recognizer implements QnAMakerRecognizer
      * @param dc The dialog context used to access state.
      */
     protected getQnAMaker(dc: DialogContext): QnAMaker {
-        const endpointKey = this.endpointKey && this.endpointKey.getValue(dc.state);
-        const hostname = this.hostname && this.hostname.getValue(dc.state);
-        const knowledgeBaseId = this.knowledgeBaseId && this.knowledgeBaseId.getValue(dc.state);
+        const { value: endpointKey, error } = this.endpointKey?.tryGetValue(dc.state);
```

Comment:
Unfortunately, destructuring throws if the right side is undefined. You need to do `const { key } = value?.value ?? {}` to be safe.